services/retentionmgr/retention.go

```diff
@@ -649,22 +656,32 @@ func (t *RetentionManager) computeRetention(job *retentionJob, log bark.Logger)
 			}).Error(`computeRetention: minAckAddr GetAckLevel failed`)
 
 			minAckAddr = store.ADDR_BEGIN
+			allHaveConsumed = false
 			break
 		}
 
+		// check if all CGs have consumed this extent
+		if ackAddr != store.ADDR_SEAL {
+			allHaveConsumed = false
+		}
+
 		// update minAckAddr, if ackAddr is less than the current value
 		if (minAckAddr == store.ADDR_END) ||
-			(minAckAddr == store.ADDR_SEAL) || // -> all existing consumers have completely consumed this extent
+			(minAckAddr == store.ADDR_SEAL) || // -> consumers we have seen so far have completely consumed this extent
 			(ackAddr != store.ADDR_SEAL && ackAddr < minAckAddr) {
 
 			minAckAddr = ackAddr
 		}
 	}
 
-	// if we were unable to find any consumer groups, set minAckAddr to ADDR_BEGIN
+	// if we were unable to find any consumer groups, set minAckAddr to
+	// ADDR_BEGIN, effectively disabling soft-retention. that said, hard
+	// retention could still be enforced.
 	if minAckAddr == store.ADDR_END {
```

Comment:
True, but that check is redundant .. if there are no consumers, the minAckAddr would be ADDR_END (the default initialization value).